std-bits/table/src/main/java/org/enso/table/data/column/builder/DateBuilder.java

```diff
@@ -45,4 +47,24 @@ public boolean accepts(Object o) {
   protected Storage<LocalDate> doSeal() {
     return new DateStorage(data, currentSize);
   }
+
+  @Override
+  public boolean canRetypeTo(StorageType type) {
+    if (Objects.equals(type, DateTimeType.INSTANCE)) {
+      return true;
+    }
+    return super.canRetypeTo(type);
+  }
```

Comment:
I understand we do this when reading Excel.

However, when reading CSV, Date and Date_Time are more distinct - I'm not convinced we should automatically apply the same logic for merging these two types as we do for Excel.

What do you think about adding `allowDateToDateTimeConversion` flag here as well? Then we could enable this logic only in Excel.

Reply:
It wouldn't have triggered for CSV as InferredBuilder has to have it turned on. However, have added the flag to DateBuilder.